### PR TITLE
fix: Incase of error, handle error

### PR DESCRIPTION
**Pull request by SpcFORK**
```shell
runtime/voxel.c: In function ‘main’:
runtime/voxel.c:34:5: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   34 |     fread(data, sizeof(char), size, fp);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### CODE REVIEW

File: runtime/voxel.c [hunk 1]
```diff
@@ -26,6 +26,13 @@ int main(int argc, char* argv[]) {
 
     fseek(fp, 0, SEEK_SET);
     fread(data, sizeof(char), size, fp);
```
Comment (James-Livesey) [inline runtime/voxel.c:28]: Looks like you forgot to remove this line — it's causing `fread` to read twice, which is causing tests to fail!

File: runtime/voxel.c [hunk 2]
```diff
@@ -26,6 +26,13 @@ int main(int argc, char* argv[]) {
...
+    
+    if (fread(data, sizeof(char), size, fp) != size) {
+        fprintf(stderr, "Error reading file content\n");
```
Comment (James-Livesey) [inline runtime/voxel.c:31]: Just me being a bit pedantic but I'd recommend adding an 's' to the end of the string to make 'content' plural — imo "Error reading file contents" reads a bit better. Happy to do this myself after merging tho!

File: runtime/voxel.c (hunk 1 above)
Comment (James-Livesey) [inline runtime/voxel.c:28]: Looks like you forgot to remove this line — it's causing `fread` to read twice, which is causing tests to fail!

**Review by James-Livesey — CHANGES_REQUESTED** [review]
Just a few minor changes that I'd recommend making, but otherwise LGTM!

File: runtime/voxel.c (hunk 1 above)
Comment (SpcFORK) [reply runtime/voxel.c:28]: Oops, sorry!
Just pasted from Replit, forgot to remove the line.

**Comment by SpcFORK** [comment]
Fixed! No more double read.

**Review by James-Livesey — APPROVED** [review]
Nice, all good from me — thanks for contributing! Will merge now :+1:

**Pull request merged**